lib/erb_lint/cli.rb

```diff
@@ -33,11 +33,12 @@ def initialize
     end
 
     def run(args = ARGV)
+      load_config
       dupped_args = args.dup
       load_options(dupped_args)
       @files = dupped_args
 
-      load_config
+      merge_config
```

Comment:
How do you feel about merging this with `load_config`? It feels like it should be "atomic".

My main concern is that the separation makes it feel like they're two separate things. On top of that, putting `load_config` at the beginning of this function makes it seem like it's important for `load_config` to be called before `load_options`.

Another option is to keep the `merge_config` method, but put it at the bottom of `load_config`. That ensures correct ordering 🙂

Reply:
It is interesting - this is a tough one to defend, but after some pretty serious debugging I feel that the loading, gathering of options, and then merging of the config are all quite separate.

On master currently, the options are gathered, then config is loaded and merged:

Options loading + config loading:
https://github.com/Shopify/erb-lint/blob/44b4b9629872bc952f76ea9f5a99fa738d0b832c/lib/erb_lint/cli.rb#L35-L40

Config merging within the load method:
https://github.com/Shopify/erb-lint/blob/44b4b9629872bc952f76ea9f5a99fa738d0b832c/lib/erb_lint/cli.rb#L150-L161

The problem I have is that now that the glob can be configured, this line needs to be changed:
https://github.com/Shopify/erb-lint/blob/44b4b9629872bc952f76ea9f5a99fa738d0b832c/lib/erb_lint/cli.rb#L256-L258

To accurately report a custom glob, I need to have the configuration loaded.
Originally I thought, oh no problem - move the `load_config` call to before the options. The problem is that the merge step inside of `load_config` can rely on options that have been passed (linters enabled etc.).

Due to the above ordering problem, I split the functions out and rearranged them.

I am absolutely open to ways to get around this, perhaps there is a better way to report the custom glob in the `option_parser`?

Thank you for your time, I realize the above is quite in depth.

Reply:
Really appreciate the thorough response, @bstaudacher. As I guessed, there _is_ an underlying reason to the ordering, it's just a little difficult to make the connection.

Now that I know about the issue, I may follow up to address this, but I really want to push your contribution through because it's super useful. I think the way to go about it is to just not show the glob in the option's, but rephrase the help in some way to mention the glob.

Again, thanks for the thoughtful response, and we really appreciate the contribution 🙌

Reply:
The error that you were getting earlier related to the RuboCop issue incase you never figured it out, likely suggests that the YAML config file for RuboCop has had syntax changed, so in this case, the rule you have set for: **Layout/AlignArguments** has had its name changed from that to this: **Layout/ArgumentAlignment.**  You can see that they are similar but the name is slightly different so the proposed solution could have been to update the rubocop configuration to refer to the correct rule you want to enable.

 RuboCop::ValidationError:
       The `Layout/AlignArguments` cop has been renamed to `Layout/ArgumentAlignment`.
       (obsolete configuration found in .erblint-rubocop20191204-16552-1p69s5q, please update it)

Figured I would chime in... :)